Dockerfile

```diff
@@ -32,32 +33,49 @@ RUN apt-get update && \
         ca-certificates \
         libreadline8 \
         python3 \
-        shared-mime-info && \
+        shared-mime-info  \
+    && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY Gemfile* package.json yarn.lock /opt/mastodon/
+
+RUN set -eux && \
     bundle config set --local deployment 'true' && \
     bundle config set --local without 'development test' && \
     bundle config set silence_root_warning true && \
     bundle install -j"$(nproc)" && \
     yarn install --pure-lockfile --network-timeout 600000
 
-FROM node:${NODE_VERSION}
+COPY --chown=mastodon:mastodon . /opt/mastodon
```

Comment:
I haven't tried running a build of this PR (I'm not a mastodon user), but have you verified that this user and group exist in this build stage? At a glance it doesn't look like they do? (you add them in the prod image though)

If they don't exist, you don't get an error about that thrown at build, it just gets set to `root:root` instead.

---

If you look at the before diff, before you shifted this line and the `RUN OTP_SECRET` below, the user had already been setup before the `USER` and `WORKDIR` lines.

That said, the `--chown` probably doesn't matter here since the only thing that happens directly after doesn't seem like it'd care, and you `COPY --chown` in your prod stage where it matters.

Reply:
Yes the chown is indeed redundant, thanks! I'm removing it
![image](https://user-images.githubusercontent.com/7402764/213869922-d0b4993f-292b-42ce-b387-4a9d20377aa8.png)